src/cpp/rtps/reader/StatefulReader.cpp

```diff
@@ -603,10 +609,15 @@ bool StatefulReader::processGapMsg(
         {
             if (pWP->irrelevant_change_set(it))
             {
-                CacheChange_t* to_remove = findCacheInFragmentedProcess(auxSN, pWP->guid());
+                CacheChange_t* to_remove = nullptr;
+                auto ret_iterator = findCacheInFragmentedProcess(auxSN, pWP->guid(), &to_remove, history_iterator);
                 if (to_remove != nullptr)
                 {
-                    mp_history->remove_change(to_remove);
+                    ret_iterator = mp_history->remove_change_nts(to_remove, ret_iterator);
```

Comment:
Perhaps it should be `history_iterator = mp_history->remove_change_nts(to_remove, ret_iterator);`

Reply:
You're right, done